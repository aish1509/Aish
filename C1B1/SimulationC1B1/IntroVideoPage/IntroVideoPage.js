import React, {useCallback} from 'react';
import {DataStore} from '@shared/core';
import IntroVideoView from '../IntroVideoView';
import {Layout} from '@math/m180';
import mockJSON from '../../../pages/mock.json';

const IntroVideoPage = () => {
	const loading = () => <div data-testid="loading">loading...</div>;
	const ready = useCallback(
		profile => (
			<IntroVideoView data-testid="IntroVideoView" data={profile} />
		),
		[]
	);
	// {provisionedBlocks: [], blockIndex: null} has been added as a mock data, once proper API implemented, This needs to be removed

	const page = ({profile}) =>
		profile
			? ready({
					...mockJSON,
					...profile
			  })
			: loading();
	return (
		<Layout>
			<DataStore.Subscribe>
				{({state}) => page(state)}
			</DataStore.Subscribe>
		</Layout>
	);
};

export default IntroVideoPage;
