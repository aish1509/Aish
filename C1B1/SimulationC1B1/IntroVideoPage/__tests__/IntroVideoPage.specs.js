import {mount, shallow} from 'enzyme';
import {shallowToJson} from 'enzyme-to-json';
import React from 'react';

import {M180Theme} from '@math/common';
import {DataStore} from '@shared/core';
import IntroVideoPage from '../index';

jest.mock('@shared/core', () => ({
	DataStore: jest.requireActual('@shared/core').DataStore,
	Logger: {
		log: jest.fn()
	}
}));
jest.mock('@math/app-m180/src/pages/mock.json');

describe('<IntroVideoPage />', () => {
	let wrapper = null;
	let mockMatch = null;

	let mockProfile = {
		username: 'username',
		scopes: 'scopes',
		app: 'm180'
	};

	describe('The page', () => {
		beforeEach(() => {
			mockMatch = {path: ''};
		});
		it('loading correctly', async () => {
			await expect(DataStore.set('profile', null)).resolves.toStrictEqual(
				null
			);
			wrapper = mount(
				<DataStore.Provider>
					<M180Theme>
						<IntroVideoPage match={mockMatch} />
					</M180Theme>
				</DataStore.Provider>
			);
			expect(wrapper.find('[data-testid="loading"]').exists()).toBe(true);
		});
		it('renders correctly', () => {
			wrapper = shallow(<IntroVideoPage match={mockMatch} />);
			expect(shallowToJson(wrapper)).toMatchSnapshot();
		});

		describe('mount component with DataStore.Provider', () => {
			beforeEach(async () => {
				await expect(
					DataStore.set('profile', mockProfile)
				).resolves.toStrictEqual(mockProfile);
				wrapper = mount(
					<DataStore.Provider>
						<M180Theme>
							<IntroVideoPage match={mockMatch} />
						</M180Theme>
					</DataStore.Provider>
				);
			});
		});
	});
});
