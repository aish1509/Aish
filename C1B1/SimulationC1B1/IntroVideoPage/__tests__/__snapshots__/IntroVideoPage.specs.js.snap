// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IntroVideoPage /> The page renders correctly 1`] = `
<Layout
  defaultStyle={
    Object {
      "display": "flex",
      "flexDirection": "column",
    }
  }
>
  <DataStoreSubscribe
    to={null}
  >
    <Component />
  </DataStoreSubscribe>
</Layout>
`;
